protocol/synthetix/test/integration/modules/core/CollateralModule/CollateralModule.deposit.test.ts

```diff
@@ -147,6 +148,32 @@ describe('CollateralModule', function () {
               });
             });
 
+            describe('when attempting to withdraw locked collateral', () => {
+              var lockedTill;
+              it('get current time', async () => {
+                const currentTime = await getTime(provider());
+                lockedTill = currentTime + 60 * 60 * 24 * 30;
+              });
+              it('reverts', async () => {
+                await systems()
+                  .Core.connect(user1)
+                  .createLock(1, Collateral.address, depositAmount, lockedTill);
+
+                // Collateral uses 6 decimals and system operates with 18,
+                // so scale up the expected amount.
+                const errorAmount = depositAmount.mul(ethers.BigNumber.from(10).pow(12));
+
+                await assertRevert(
+                  systems().Core.connect(user1).withdraw(1, Collateral.address, depositAmount),
+                  `InsufficientAccountCollateral("${errorAmount}")`,
+                  systems().Core
+                );
+              });
+              it('unlock collateral', async () => {
```

Comment:
this test has no conditions validating that the collateral was actually unlocked

also this should go in a nested `describe` with a `before` section running the fast forward

Reply:
(If you did not fast-forward, then the following withdraw would have failed)